src/app/code/community/Omise/Gateway/Model/Payment/Creditcard.php

```diff
@@ -74,12 +74,12 @@ public function initialize($payment_action, $state_object)
 
             case Mage_Payment_Model_Method_Abstract::ACTION_AUTHORIZE_CAPTURE:
                 $invoice = $order->prepareInvoice()->register();
-
+                $invoice->setState(Mage_Sales_Model_Order_Invoice::STATE_PAID)->save();
```

Comment:
Just wonder why we set the invoice's state to `paid` before the actual charge has been created?

Reply:
@guzzilar Look like it should be the way it was before.

Reply:
Doesn't seem right. If something goes wrong during charge creation then this will create issues?